### ISSUE TITLE
Selectable setting for default file removal behaviour

### ISSUE THREAD

**Issue by timmersr** [nextcloud/android#3342]
I am always a bit anxious when removing a downloaded file from my phone, because I have to select 'only local' in between two other options, in order to leave it on the server.

I am only looking to remove it from my phone, and never from the server.

Could you create this as selectable setting for default removals?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @timmersr I strongly vote no. 

We do have the trashbin so the files (if deleted accidentially from the server) can be restored and I wouldn't want to start implementing UI flag to have a customizable UI on a per-user level available since this doesn't add real value to the app but a lot of complexity that needs to be maintained in the future.

**Comment by tobiasKaminsky** [comment]
![2018-12-11-110514](https://user-images.githubusercontent.com/5836855/49793098-a8633200-fd34-11e8-9c02-cdcf770c1714.png)

I fully agree with Andy. 
What we can do is to highlight "local only" as this is probably the most used option, and the other one in grey.

**Comment by tobiasKaminsky** [comment]
@nextcloud/designers

**Comment by tobiasKaminsky** [comment]
We had also the idea to offer an undo on deletion as snackbar. 
This will give the user a bit of confidence that nothing really can go wrong…

**Comment by jancborchardt** [comment]
Right, this is a good point to raise, I also really would like to simplify this dialog. Just to start, no we will not add a setting for this.

> What we can do is to highlight "local only" as this is probably the most used option, and the other one in grey.

We don’t have a way to really make an informed decision here through metrics, right? Because when I delete, I basically always mean _delete_ from the server. Don’t care about phone space even though it’s not huge storage.

This is also how Google Drive and Dropbox do it. They don’t have a specific "Local only" deletion. Files simply get removed, and you get a Snackbar confirmation "`Filename` deleted  [ Undo ]". It’s seamless and not scary, and since we have deleted files, you can restore them at any point too.

So for people who like to keep an eye on their phone storage, what we could do is have an entry somewhere to clear the downloaded files? Probably in the settings. Then again, Android offers these ways of cleaning data of apps.

**Comment by tobiasKaminsky** [comment]
> This is also how Google Drive and Dropbox do it.

This is because they have all files only available in cache. They do not expose the downloaded files the way we are doing it. So we cannot directly compare it.



> So for people who like to keep an eye on their phone storage, what we could do is have an entry somewhere to clear the downloaded files?

What if you want to delete only a part of your downloaded files? E.g. remove all downloaded music, but keep the vacation pictures?



> Android offers these ways of cleaning data of apps.

We already have this, but it deletes all files from all accounts…

**Comment by jancborchardt** [comment]
> This is because they have all files only available in cache. They do not expose the downloaded files the way we are doing it. So we cannot directly compare it.

Ok, but that’s just how it’s implemented. From the user perspective this doesn’t matter that much.


> What if you want to delete only a part of your downloaded files? E.g. remove all downloaded music, but keep the vacation pictures?

Good point. So then we should have a thing in the settings called

> **Clear downloaded files**
> Choose which to keep and which to remove

And it will show basically a folder-picker dialog with checkboxes where by default everything is selected, and two buttons at the bottom "Remove local files" and "Cancel". And there you can unselect what you like.

What do you think?

**Comment by joshtrichards** [comment]
> Good point. So then we should have a thing in the settings called

> > Clear downloaded files
> > Choose which to keep and which to remove

> What do you think?

It might make sense to put this sort of thing under the existing *On Device* tab since that's where all the downloaded files are already centralized.

> I am always a bit anxious when removing a downloaded file from my phone, because I have to select 'only local' in between two other options, in order to leave it on the server.

At least today this is somewhat alleviated because *Delete* is now a distinct button visually distinguishing it from the adjacent *Local only*. Also, as already noted, there's the trash bin so accidental deletions are not immediately permanent (though I see how anxiousness might continue since you may not discover the evidence of what button you pushed until weeks/months later).

Personally I find the shifting of the  *Keep* option on the Delete dialog for downloaded files jarring:

![Screenshot_20231219-153701.png](https://github.com/nextcloud/android/assets/1731941/82a527d0-ad97-45b5-9c8c-147b1a12d910)

![Screenshot_20231219-153921.png](https://github.com/nextcloud/android/assets/1731941/c8590bb8-5d0c-4003-8e79-a8ba224cd130)

I'd prefer the *Keep* action stayed in the same place for either file type so the UI is always consistent and muscle memory friendly, and *Local only* gets added at the far left for Downloaded files instead of where *Keep* gets shifted to for those.

It also would have the benefit of *Local only* no longer being stuck being two other options - the original concern brought up in this Issue.
